### PR TITLE
chore: replace deprecated String.prototype.substr()

### PR DESCRIPTION
**Pull request by CommanderRoot**
**Summary**

[String.prototype.substr()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr) is deprecated so we replace it with [String.prototype.slice()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) which works similarily but isn't deprecated.
.substr() probably isn't going away anytime soon but the change is trivial so it doesn't hurt to do it.

**Test plan**

I have checked the return of each statement to make sure it's the same. All test are OK as well.

**Checklist**

Please add a `x` inside each checkbox:

- [x] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [x] Code is formatted via running `yarn format`.
- [x] Tests are passing via running `yarn test`.
- [ ] The status checks are successful (continuous integration). Those can be seen below.

**A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by erezrokah — APPROVED** [review]
Thanks @CommanderRoot! I followed up with a lint rule for this, and some more changes.

Please let me know what you think.

File: packages/netlify-cms-core/src/formats/frontmatter.ts [hunk 1]
```diff
@@ -27,15 +27,16 @@ const parsers = {
...
       }
       return jsonFormatter.fromFile(JSONinput);
     },
     stringify: (metadata: object) => {
       let JSONoutput = jsonFormatter.toFile(metadata).trim();
       // Trim leading and trailing brackets.
-      if (JSONoutput.substr(0, 1) === '{' && JSONoutput.substr(-1) === '}') {
+      if (JSONoutput.slice(0, 1) === '{' && JSONoutput.slice(-1) === '}') {
+        // eslint-disable-next-line unicorn/prefer-string-slice
         JSONoutput = JSONoutput.substring(1, JSONoutput.length - 1);
```
Comment (CommanderRoot) [inline packages/netlify-cms-core/src/formats/frontmatter.ts:40]: JSONoutput = JSONoutput.substring(1, JSONoutput.length - 1);
can be
JSONoutput = JSONoutput.slice(1, -1);

File: packages/netlify-cms-core/src/formats/frontmatter.ts [hunk 2]
```diff
@@ -130,6 +132,7 @@ export class FrontmatterFormatter {
       comments,
       ...format,
     });
+    // eslint-disable-next-line unicorn/prefer-string-slice
     return trimLastLineBreak && file.slice(-1) === '\n' ? file.substring(0, file.length - 1) : file;
```
Comment (CommanderRoot) [inline packages/netlify-cms-core/src/formats/frontmatter.ts:136]: `file.substring(0, file.length - 1)`
can be
`file.slice(0, -1)`

File: packages/netlify-cms-lib-util/src/APIUtils.ts [hunk 3]
```diff
@@ -26,11 +26,11 @@ export function generateContentKey(collectionName: string, slug: string) {
 
 export function parseContentKey(contentKey: string) {
   const index = contentKey.indexOf('/');
-  return { collection: contentKey.substr(0, index), slug: contentKey.substr(index + 1) };
+  return { collection: contentKey.slice(0, index), slug: contentKey.slice(index + 1) };
```
Comment (CommanderRoot) [inline packages/netlify-cms-lib-util/src/APIUtils.ts:29]: `index` could possibly be negative (-1) so to have the same result as `substr()` we either have to check for that or use `substring` on `contentKey.slice(0, index)`

File: packages/netlify-cms-widget-markdown/src/serializers/slateRemark.js [hunk 4]
```diff
@@ -177,7 +177,9 @@ export default function slateToRemark(raw, { voidCodeBlock }) {
     const index = node.text.search(exp);
     if (index > -1) {
       const substringIndex = trailing ? index : index + 1;
+      // eslint-disable-next-line unicorn/prefer-string-slice
       const firstSplit = node.text.substring(0, substringIndex);
+      // eslint-disable-next-line unicorn/prefer-string-slice
```
Comment (CommanderRoot) [inline packages/netlify-cms-widget-markdown/src/serializers/slateRemark.js:182]: Index is never negative so we can use slice here

**Comment by CommanderRoot** [comment]
I've added a few comments in regards to optimisation / more slice usage. Other than that it looks good to me

File: packages/netlify-cms-lib-util/src/APIUtils.ts (hunk 3 above)
Comment (erezrokah) [reply packages/netlify-cms-lib-util/src/APIUtils.ts:29]: It's expected for `contentKey` to always include a `/` (it's the branch name, e.g. `cms/...`

**Comment by erezrokah** [comment]
@CommanderRoot looks like our bot merge this... Would you like to follow up with your suggestions?

**Comment by CommanderRoot** [comment]
Sure, I have created #6352 which removes all uses of `substring()` in this repo

**Pull request merged**